iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/extensions/SlackTerminalXmlSerializer.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.xml.extensions;
+
+import com.google.auto.service.AutoService;
+import com.powsybl.commons.extensions.AbstractExtensionXmlSerializer;
+import com.powsybl.commons.extensions.ExtensionXmlSerializer;
+import com.powsybl.commons.xml.XmlReaderContext;
+import com.powsybl.commons.xml.XmlUtil;
+import com.powsybl.commons.xml.XmlWriterContext;
+import com.powsybl.iidm.network.Terminal;
+import com.powsybl.iidm.network.VoltageLevel;
+import com.powsybl.iidm.network.extensions.SlackTerminal;
+import com.powsybl.iidm.network.extensions.SlackTerminalAdder;
+import com.powsybl.iidm.xml.NetworkXmlReaderContext;
+import com.powsybl.iidm.xml.NetworkXmlWriterContext;
+import com.powsybl.iidm.xml.TerminalRefXml;
+
+import javax.xml.stream.XMLStreamException;
+
+/**
+ * @author Florian Dupuy <florian.dupuy at rte-france.com>
+ */
+@AutoService(ExtensionXmlSerializer.class)
+public class SlackTerminalXmlSerializer extends AbstractExtensionXmlSerializer<VoltageLevel, SlackTerminal> {
+
+    private static final String ELEM_TERMINAL_REF = "terminalRef";
+
+    public SlackTerminalXmlSerializer() {
+        super("slackTerminal", "network", SlackTerminal.class, true, "slackTerminal.xsd",
+            "http://www.powsybl.org/schema/iidm/ext/slack_terminal/1_0", "slt");
+    }
+
+    @Override
+    public void write(SlackTerminal slackTerminal, XmlWriterContext context) throws XMLStreamException {
+        TerminalRefXml.writeTerminalRef(slackTerminal.getTerminal(), (NetworkXmlWriterContext) context, getNamespaceUri(), ELEM_TERMINAL_REF);
+    }
```

Comment:
I would check if the terminal is not null. I would also override the `hasSubElements`: this method should return true if the terminal is set, false otherwise

Reply:
Corrected. And hasSubElements is now always false, as the attributes are now directly in the slackTerminal (see comment below)